### PR TITLE
front: Add Turtle component

### PR DESCRIPTION
**Pull request by helaan**

### CODE REVIEW

File: front/src/Turtle.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  flex: 1;
+`;
+
+const DefaultTurtle = styled.div`
+  width: 90%;
+  margin-left: auto;
+  margin-right: auto;
+  margin-top: 2px;
+  margin-bottom: 2px;
+  padding: 20px;
```
Comment (satanja) [inline front/src/Turtle.js]: `2rem`

File: front/src/Turtle.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  border-style: solid;
+  border: 3px 3px 3px 3px;
```
Comment (satanja) [inline front/src/Turtle.js]: In terms of `rem`, I think `0.25rem` is okay. Slightly thicker border though.

File: front/src/Turtle.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  font-size: 4vmin;
```
Comment (satanja) [inline front/src/Turtle.js]: `0.25rem`

File: front/src/Turtle.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
+import PropTypes from "prop-types";
+import React from "react";
+import Dropdown from "./Dropdown";
+import Battery from "./Battery";
+import styled from "styled-components";
+
+const TEAM_VALUES = ["Magenta", "Cyan"];
+const HOME_VALUES = ["Yellow home", "Blue home"];
+const ROLE_VALUES = [
+  "INACTIVE",
```
Comment (g868) [inline front/src/Turtle.js]: Naming is not consistent, but this is due to the Tech United naming (it was copied)

File: front/src/Turtle/index.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+const DefaultTurtle = styled.div`
+  width: 90%;
+  margin: 0.2rem auto;
+  padding: 2rem;
+  border-style: solid;
+  border: 0.25rem;
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  font-size: 4vmin;
```
Comment (bafonins) [inline front/src/Turtle/index.js]: `vmin`?

File: front/src/Turtle/index.js [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+ * - editable: whether this turtle's properties can be edited
+ * - onChange: a function with two arguments that is called when one of the dropdowns is changed. The first argument is name of the prop that is changed, the second argument is its new value.
+ */
+const Turtle = props => {
+  const { battery, home, id, role, team } = props.turtle;
+  const { editable, onChange } = props;
+  return (
+    <DefaultTurtle>
+      <Battery percentage={battery} />
+      <SubSection>
```
Comment (bafonins) [inline front/src/Turtle/index.js]: Do you really need this `div` ?

Comment (helaan) [reply front/src/Turtle/index.js]: No, but there might be additional indicators in the middle. We actually plan on doing that, but these indicators weren't in the current design

File: front/src/Turtle/index.js [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+ * Show all details for a turtle
+ * Author: H.E. van der Laan
+ *
+ * Props:
+ * - turtle: An object containing the following turtle details:
+ *   - battery: the current battery status of the turtle
+ *   - home: the current home goal of this robot
+ *   - id: the ID number of the turtle
+ *   - role: the current role of this robot
+ *   - team: the current team of this robot
```
Comment (g868) [inline front/src/Turtle/index.js]: We should be consistent and keep using "turtle" instead of "robot"

**Pull request merged**